src/vision/status_quo/grace_wants_a_zero_copy_api.md

```diff
@@ -0,0 +1,78 @@
+# 😱 Status quo stories: Grace wants a zero-copy API
+
+[Alan]: ../characters/alan.md
+[Grace]: ../characters/grace.md
+[Niklaus]: ../characters/niklaus.md
+[Barbara]: ../characters/barbara.md
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today.
+
+## The story
+
+Grace had written lots of operating system code in the past, and up until recently was working on a project using [DPDK](https://www.dpdk.org/) for zero-copy networking. The vast majority of the bugs
+that Grace found were related to memory (mis)management, so she is excited for the prospect of trying Rust as part of her new job.
+
+However, Grace has a hard time getting this to work without heavily resorting to `unsafe` constructs. As she evolves her undertanding of Rust, she looks hopefully at the signature of `poll_read`:
+
+```ignore
```

Comment:
All the code blocks should be changed. We don't run the code samples anyway, but this will ensure proper syntax highlighting 
```suggestion
```rust
```